src/app_service/service/contacts/service.nim

```diff
@@ -139,11 +139,16 @@ QtObject:
     try:
       let response = status_contacts.getContacts()
 
-      let contacts = map(response.result.getElems(), proc(x: JsonNode): ContactsDto = x.toContactsDto())
+      let contacts = map(response.result.getElems(), proc(x: JsonNode): ContactDto = x.toContactsDto())
 
       for contact in contacts:
-        self.addContact(self.constructContactDetails(contact))
-
+        var c = contact
+        if contact.displayName == "":
+          # Fetch the contact info if we do not have it
+          self.requestContactInfo(contact.id)
+          # Set fetching to true so the UI can show a loader if needed
+          c.fetching = true
```

Comment:
> I don't see where it is set to false thinking Is it set to false on each contact update through toContactsDto? If that's the case, then I am afraid the loading state will be way too unpredictable.

It's set to false if `requestContactInfo` fails. Also, kinda like you pointed out, any contact that comes from status-go is passed through `toContactsDto`, and it will not contain `fetching`, so it will be set to false again. So at the end of `requestContactInfo`, whether or not it succeeded in finding the user's info.

>  If that's the case, then I am afraid the loading state will be way too unpredictable.

yeah, I know that it's a bit unpredictable, but at the same time, it can't be worse than what we have right now.

Right now, the users with unknown display names show the 3 word name and letter identicon all the time.
This change makes it so that at the start, for max 15 seconds (that's the timeout for `requestContactInfo`), we show a loader, and if we find the info, the loader is replaced by the right info.
If we do not find it, the 3 word name is shown again.
There is the edge case where some random signal from the user comes in before, but that's also a success, because that would replace the loading by the right info.

So all in all, even though `fetching` might be reset to `false` preemptively, I don't think it's a big issue.

>  Also, I believe the logic of fetching missing contacts is done by status-go seamlessly as it subscribes to contacts' topics, meaning it will fetch store nodes on startup and eventually fill missing contacts data (although, I would need to double check that).

I thought it didn't, because we have some many missing contact infos (3 word name in multiple places), but maybe it's just because we can't fetch it because it's been more than 9 days.

We need to double check. If that's the case, then you're right, I probably did all of this for no reason :cry: